tests/api_fastapi/execution_api/routes/test_task_instances.py

```diff
@@ -193,6 +195,50 @@ def test_ti_update_state_database_error(self, client, session, create_task_insta
             assert response.status_code == 500
             assert response.json()["detail"] == "Database error occurred"
 
+    def test_ti_update_state_to_deferred(self, client, session, create_task_instance):
+        """
+        Test that tests if the transition to deferred state is handled correctly.
+        """
+        clear_db_runs()
+
+        ti = create_task_instance(
+            task_id="test_ti_update_state_to_deferred",
+            state=State.RUNNING,
+            session=session,
+        )
+        session.commit()
+
+        payload = {
+            "state": "deferred",
+            "classpath": "my-class-path",
+            "kwargs": {},
+            "created_date": "2024-10-31T12:00:00Z",
```

Comment:
Hmm, why is this being sent in the payload? Should it be the time the server received the request instead?

Reply:
It is handled this way:
```
self.created_date = created_date or timezone.utcnow()
```

In case we do not pass it, it will take the UTC now. Should we allow the option to override it or remove it entirely?

Reply:
As per kaxil's comment below, removed this logic